### PR TITLE
Suggestion: ALL_THIRD, Both_Accessory, Both_Hand in item_db.yml

### PR DESCRIPTION
**Pull request by Atemo**
* **Addressed Issue(s)**: -

* **Server Mode**: -

* **Description of Pull Request**: 

Simple suggestion to combine 
- the 3 types of third classes in a mask
- Right_Accessory + Left_Accessory = Both_Accessory
- Right_Hand + Left_Hand = Both_Hand

### CODE REVIEW

**Review by Daegaladh — APPROVED** [review]

File: src/map/itemdb.hpp [hunk 1]
```diff
@@ -225,6 +225,7 @@ enum e_item_job : uint16
 	ITEMJ_THIRD       = 0x08,
 	ITEMJ_THIRD_UPPER = 0x10,
 	ITEMJ_THIRD_BABY  = 0x20,
+	ITEMJ_ALL_THIRD   = 0x38,
```
Comment (Lemongrass3110) [inline src/map/itemdb.hpp:228]:
```suggestion

```

File: src/map/itemdb.hpp [hunk 2]
```diff
@@ -225,6 +225,7 @@ enum e_item_job : uint16
...
 	ITEMJ_MAX         = 0xFF,
```
Comment (Lemongrass3110) [inline src/map/itemdb.hpp:230]:
```suggestion

	ITEMJ_ALL_UPPER = ITEMJ_UPPER | ITEMJ_THIRD_UPPER,
	ITEMJ_ALL_BABY = ITEMJ_BABY | ITEMJ_THIRD_BABY,
	ITEMJ_ALL_THIRD = ITEMJ_THIRD | ITEMJ_THIRD_UPPER | ITEMJ_THIRD_BABY,

```

File: doc/item_db.txt [hunk 3]
```diff
@@ -155,6 +155,7 @@ Baby        - Baby classes (no Third-Baby classes).
 Third       - Third classes (no Transcedent-Third or Third-Baby classes).
 Third_Upper - Transcedent-Third classes.
 Third_Baby  - Third-Baby classes.
+All_Third   - Applies to all Third classes.
```
Comment (Lemongrass3110) [inline doc/item_db.txt:158]:
```suggestion
All_Upper   - All Transcedent classes
All_Baby    - All baby classes
All_Third   - Applies to all Third classes.
```

File: src/map/script_constants.hpp [hunk 4]
```diff
@@ -7965,6 +7967,7 @@
 	export_constant(ITEMJ_THIRD);
 	export_constant(ITEMJ_THIRD_UPPER);
 	export_constant(ITEMJ_THIRD_BABY);
+	export_constant(ITEMJ_ALL_THIRD);
```
Comment (Lemongrass3110) [inline src/map/script_constants.hpp:7970]:
```suggestion
	export_constant(ITEMJ_ALL_UPPER);
	export_constant(ITEMJ_ALL_BABY);
	export_constant(ITEMJ_ALL_THIRD);
```

File: src/map/script_constants.hpp [hunk 5]
```diff
@@ -813,8 +813,10 @@
 	export_constant(EQP_ACC_RL);
 	export_constant(EQP_SHADOW_ACC_RL);
 
+	export_constant2("EQP_Both_Hand", (EQP_HAND_R|EQP_HAND_L));
```
Comment (Lemongrass3110) [inline src/map/script_constants.hpp:816]: Do not ask me why, but there are some additional EQP defines inside pc.hpp

This should use: EQP_ARMS

**Review by Lemongrass3110 — CHANGES_REQUESTED** [review]
Can you please add the code for All_Upper & All_Baby to yaml2sql? Would have been a pain to do this on Github sorry...

Also it  might be good to integrate these changes into csv2yaml?

Are you going to update the database in a seperate pull/commit, after this has been merged?

**Comment by Atemo** [comment]
>Also it might be good to integrate these changes into csv2yaml?
Are you going to update the database in a seperate pull/commit, after this has been merged?

Yes.
I will update the DB in this PR.

**Review by Lemongrass3110 — APPROVED** [review]

**Comment by jl1990** [comment]
This is causing issues:

`[Error]: Occurred in file 'db/re/item_db_usable.yml' on line 11886 and column 17.
[Error]: Invalid class upper All_Third, defaulting to All.
[Error]: Occurred in file 'db/re/item_db_usable.yml' on line 43530 and column 17.
[Error]: Invalid class upper All_Third, defaulting to All.
[Error]: Occurred in file 'db/re/item_db_usable.yml' on line 46327 and column 17.
[Status]: Done reading '3432' entries in 'db/re/item_db_usable.yml'
[Error]: Invalid equip location Both_Hand, defaulting to IT_ETC.
[Error]: Occurred in file 'db/re/item_db_equip.yml' on line 506 and column 17.
[Error]: Item type is not a weapon, defaulting to 0.
[Error]: Occurred in file 'db/re/item_db_equip.yml' on line 507 and column 17.
[Error]: Invalid equip location Both_Hand, defaulting to IT_ETC.
[Error]: Occurred in file 'db/re/item_db_equip.yml' on line 525 and column 17.
[Error]: Item type is not a weapon, defaulting to 0.
[Error]: Occurred in file 'db/re/item_db_equip.yml' on line 526 and column 17.
`

**Pull request merged**